### PR TITLE
[Veo] Add new extractor

### PR DESCRIPTION
**Pull request by i6t**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

https://github.com/ytdl-org/youtube-dl/issues/29445

### CODE REVIEW

File: yt_dlp/extractor/veo.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        view_count = int_or_none(metadata.get('view_count'))
+
+        formats = []
+        for fmt in video_data:
+            mimetype = fmt.get('mime_type')
+            # skip configuration file for panoramic video
+            if mimetype == 'video/mp2t':
+                continue
+            height = int_or_none(fmt.get('height'))
+            bitrate = int_or_none(fmt.get('bit_rate'))
```
Comment (pukkandan) [inline yt_dlp/extractor/veo.py:53]:
```suggestion
            bitrate = int_or_none(fmt.get('bit_rate'), scale=1000)
```
`bitrate / 1000 if bitrate else None` below can be removed

File: yt_dlp/extractor/veo.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+class VeoIE(InfoExtractor):
+    _VALID_URL = r'https?://app\.veo\.co/matches/(?P<id>[0-9A-Za-z-]+)'
+
+    _TEST = {
+        'url': 'https://app.veo.co/matches/20201027-last-period/',
+        'info_dict': {
+            'id': '20201027-last-period',
+            'ext': 'mp4',
+            'title': 'Akidemy u11s v Bradford Boys u11s (Game 3)',
+            'thumbnail': 'https://c.veocdn.com/8733223a-c6a0-4947-a309-f0f9fb44a1e4/standard/machine/8011e457/thumbnail.jpg',
```
Comment (pukkandan) [inline yt_dlp/extractor/veo.py:23]:
```suggestion
            'thumbnail': 're:https://c.veocdn.com/.+/thumbnail.jpg',
```
no need to hardcode the url

File: yt_dlp/extractor/veo.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    mimetype2ext,
+    unified_timestamp,
+    url_or_none,
+)
+
+
+class VeoIE(InfoExtractor):
+    _VALID_URL = r'https?://app\.veo\.co/matches/(?P<id>[0-9A-Za-z-]+)'
+
+    _TEST = {
```
Comment (pukkandan) [inline yt_dlp/extractor/veo.py:17]: even though there is only single test, it is recommended to use `_TESTS`

File: yt_dlp/extractor/veo.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+from ..utils import (
+    int_or_none,
+    mimetype2ext,
+    unified_timestamp,
+    url_or_none,
+)
+
+
+class VeoIE(InfoExtractor):
+    _VALID_URL = r'https?://app\.veo\.co/matches/(?P<id>[0-9A-Za-z-]+)'
```
Comment (fvogel) [inline yt_dlp/extractor/veo.py:15]: Veo URL can have underscores. The current regex does not catch them resulting in a 404 error when trying to fetch the video JSON.

Example of a faulty VEO URL: [https://app.veo.co/matches/20220313-2022-03-13_u15m-plsjq-vs-csl/](https://app.veo.co/matches/20220313-2022-03-13_u15m-plsjq-vs-csl/)

```suggestion
    _VALID_URL = r'https?://app\.veo\.co/matches/(?P<id>[0-9A-Za-z-_]+)'
```

**Pull request merged**